### PR TITLE
DROOLS-2648 DMN v1.2 weekday on 'date', 'date and time'

### PR DESCRIPTION
**Pull request by tarilabs**
/cc @etirelli @kurobako @baldimir @winklerm

### CODE REVIEW

**Comment by tarilabs** [comment]
same as weekend PRs
```
11:45:21 [INFO] ------------------------------------------------------------------------
11:45:21 [INFO] Reactor Summary:
11:45:21 [INFO] 
11:45:21 [INFO] Drools Multiproject ................................ SUCCESS [  4.082 s]
11:45:21 [INFO] KIE :: Test Utility Classes ........................ SUCCESS [ 22.561 s]
11:45:21 [INFO] Drools :: Core ..................................... SUCCESS [02:39 min]
11:45:21 [INFO] Drools :: Compiler ................................. SUCCESS [03:38 min]
11:45:21 [INFO] Drools :: Beliefs .................................. SUCCESS [ 24.705 s]
11:45:21 [INFO] Drools :: CDI ...................................... SUCCESS [ 34.793 s]
11:45:21 [INFO] Drools :: Verifier ................................. SUCCESS [01:26 min]
11:45:21 [INFO] Drools Persistence projects ........................ SUCCESS [  0.775 s]
11:45:21 [INFO] Drools :: Persistence :: API ....................... SUCCESS [ 16.792 s]
11:45:21 [INFO] Drools :: Persistence :: JPA ....................... SUCCESS [ 56.963 s]
11:45:21 [INFO] Drools :: Templates ................................ SUCCESS [ 23.139 s]
11:45:21 [INFO] Drools :: Decision Tables .......................... SUCCESS [ 23.702 s]
11:45:21 [INFO] Drools examples .................................... SUCCESS [ 38.289 s]
11:45:21 [INFO] KIE :: PMML - Compiler ............................. SUCCESS [03:10 min]
11:45:21 [INFO] Drools :: Scorecards ............................... SUCCESS [02:04 min]
11:45:21 [INFO] KIE :: CI .......................................... SUCCESS [04:04 min]
11:45:21 [INFO] KIE :: CI for OSGi ................................. SUCCESS [ 13.949 s]
11:45:21 [INFO] KIE :: Decision Model Notation ..................... SUCCESS [  0.742 s]
11:45:21 [INFO] KIE :: Decision Model Notation :: Model ............ SUCCESS [ 21.154 s]
11:45:21 [INFO] KIE :: Decision Model Notation :: API .............. SUCCESS [ 12.982 s]
11:45:21 [INFO] KIE :: Decision Model Notation :: FEEL ............. SUCCESS [ 46.585 s]
11:45:21 [INFO] KIE :: Decision Model Notation :: Backend .......... SUCCESS [ 25.754 s]
11:45:21 [INFO] KIE :: Decision Model Notation :: Core ............. SUCCESS [01:20 min]
11:45:21 [INFO] KIE :: Decision Model Notation :: Validation ....... SUCCESS [ 47.918 s]
11:45:21 [INFO] KIE :: Decision Model Notation :: TCK .............. SUCCESS [  0.491 s]
11:45:21 [INFO] KIE :: Decision Model Notation :: Signavio extensions support SUCCESS [ 22.418 s]
11:45:21 [INFO] Drools API examples ................................ SUCCESS [  0.355 s]
11:45:21 [INFO] Drools API examples - Default KieSession ........... SUCCESS [ 17.590 s]
11:45:21 [INFO] Drools API examples - Named KieSession ............. SUCCESS [ 11.449 s]
11:45:21 [INFO] Drools API examples - KieBase Inclusion ............ SUCCESS [ 10.789 s]
11:45:21 [INFO] Drools API examples - Default KieSession From File . SUCCESS [ 11.958 s]
11:45:21 [INFO] Drools API examples - Named KieSession from File ... SUCCESS [ 12.140 s]
11:45:21 [INFO] Drools API examples - KieModule from Multiple Files  SUCCESS [ 15.374 s]
11:45:21 [INFO] Drools API examples - KieFileSystemExample ......... SUCCESS [ 14.206 s]
11:45:21 [INFO] Drools API examples - KieModuleModel Example ....... SUCCESS [ 15.153 s]
11:45:21 [INFO] Drools API examples - KieContainer from KieRepo .... SUCCESS [ 22.754 s]
11:45:21 [INFO] Drools API examples - Multiple KieBases ............ SUCCESS [  3.536 s]
11:45:21 [INFO] Drools API examples - Reactive KieSession .......... SUCCESS [ 12.557 s]
11:45:21 [INFO] Drools API examples - RuleUnit ..................... SUCCESS [ 15.001 s]
11:45:21 [INFO] Drools CDI examples ................................ SUCCESS [  0.717 s]
11:45:21 [INFO] Drools CDI Example ................................. SUCCESS [ 13.387 s]
11:45:21 [INFO] Drools CDI Example With Inclusion .................. SUCCESS [ 14.277 s]
11:45:21 [INFO] cdi-example-scopes ................................. SUCCESS [ 18.153 s]
11:45:21 [INFO] Drools Workbench - Models .......................... SUCCESS [  0.248 s]
11:45:21 [INFO] Drools Workbench - Data Model API .................. SUCCESS [ 27.785 s]
11:45:21 [INFO] Drools Workbench - Common Model .................... SUCCESS [ 29.573 s]
11:45:21 [INFO] Drools Workbench - Guided Decision Table Model ..... SUCCESS [ 26.199 s]
11:45:21 [INFO] Drools Workbench - Guided Decision Tree Model ...... SUCCESS [ 19.555 s]
11:45:21 [INFO] Drools Workbench - Guided Rule Templates Model ..... SUCCESS [ 26.995 s]
11:45:21 [INFO] Drools Workbench - Guided Score Cards Model ........ SUCCESS [ 46.729 s]
11:45:21 [INFO] Drools Workbench - Test Scenarios Model ............ SUCCESS [ 22.735 s]
11:45:21 [INFO] drools-model ....................................... SUCCESS [  0.361 s]
11:45:21 [INFO] drools-canonical-model ............................. SUCCESS [ 32.244 s]
11:45:21 [INFO] drools-model-compiler .............................. SUCCESS [02:26 min]
11:45:21 [INFO] Drools :: Test Coverage ............................ SUCCESS [  0.714 s]
11:45:21 [INFO] Drools :: Test Coverage :: Standalone reproducers .. SUCCESS [  0.491 s]
11:45:21 [INFO] Drools :: Test Coverage :: KIE-CI with Domain classes :: Parent SUCCESS [  0.243 s]
11:45:21 [INFO] Drools :: Test Coverage :: KIE-CI with Domain classes :: Domain SUCCESS [ 12.934 s]
11:45:21 [INFO] Drools :: Test Coverage :: KIE-CI with Domain classes :: KJar SUCCESS [  1.880 s]
11:45:21 [INFO] Drools :: Test Coverage :: KIE-CI with Domain classes :: Tests SUCCESS [ 15.529 s]
11:45:21 [INFO] Drools :: Test Coverage :: KIE-CI without Domain classes :: Parent SUCCESS [  0.195 s]
11:45:21 [INFO] Drools :: Test Coverage :: KIE-CI without Domain classes :: Domain SUCCESS [ 10.778 s]
11:45:21 [INFO] Drools :: Test Coverage :: KIE-CI without Domain classes :: KJar SUCCESS [  2.427 s]
11:45:21 [INFO] Drools :: Test Coverage :: KIE-CI without Domain classes :: Tests SUCCESS [ 11.146 s]
11:45:21 [INFO] Drools :: Test Coverage :: Test Suite .............. SUCCESS [11:07 min]
11:45:21 [INFO] Drools :: Test Coverage :: Compiler integration tests SUCCESS [02:51 min]
11:45:21 [INFO] Drools distribution ................................ FAILURE [  1.734 s]
11:45:21 [INFO] ------------------------------------------------------------------------
11:45:21 [INFO] BUILD FAILURE
11:45:21 [INFO] ------------------------------------------------------------------------
11:45:21 [INFO] Total time: 25:58 min (Wall Clock)
11:45:21 [INFO] Finished at: 2018-06-25T05:45:21-04:00
11:45:23 [INFO] Final Memory: 600M/1891M
11:45:23 [INFO] ------------------------------------------------------------------------
11:45:23 [WARNING] The requested profile "wildfly11" could not be activated because it does not exist.
11:45:23 [ERROR] Failed to execute goal on project drools-distribution: Could not resolve dependencies for project org.drools:drools-distribution:pom:7.8.0-SNAPSHOT: The following artifacts could not be resolved: org.kie:kie-api:jar:javadoc:7.8.0-SNAPSHOT, org.kie:kie-internal:jar:javadoc:7.8.0-SNAPSHOT: Could not find artifact org.kie:kie-api:jar:javadoc:7.8.0-SNAPSHOT in jboss-public-repository-group (https://repository.jboss.org/nexus/content/groups/public/) -> [Help 1]
11:45:23 org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal on project drools-distribution: Could not resolve dependencies for project org.drools:drools-distribution:pom:7.8.0-SNAPSHOT: The following artifacts could not be resolved: org.kie:kie-api:jar:javadoc:7.8.0-SNAPSHOT, org.kie:kie-internal:jar:javadoc:7.8.0-SNAPSHOT: Could not find artifact org.kie:kie-api:jar:javadoc:7.8.0-SNAPSHOT in jboss-public-repository-group (https://repository.jboss.org/nexus/content/groups/public/)
11:45:23     at org.apache.maven.lifecycle.internal.LifecycleDependencyResolver.getDependencies (LifecycleDependencyResolver.java:249)
11:45:23     at org.apache.maven.lifecycle.internal.LifecycleDependencyResolver.resolveProjectDependencies (LifecycleDependencyResolver.java:145)
11:45:23     at org.apache.maven.lifecycle.internal.MojoExecutor.ensureDependenciesAreResolved (MojoExecutor.java:246)
11:45:23     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:200)
11:45:23     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
11:45:23     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
11:45:23     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
11:45:23     at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call (MultiThreadedBuilder.java:200)
11:45:23     at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call (MultiThreadedBuilder.java:196)
11:45:23     at java.util.concurrent.FutureTask.run (FutureTask.java:266)
11:45:23     at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:511)
11:45:23     at java.util.concurrent.FutureTask.run (FutureTask.java:266)
11:45:23     at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1149)
11:45:23     at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:624)
11:45:23     at java.lang.Thread.run (Thread.java:748)
11:45:23 Caused by: org.apache.maven.project.DependencyResolutionException: Could not resolve dependencies for project org.drools:drools-distribution:pom:7.8.0-SNAPSHOT: The following artifacts could not be resolved: org.kie:kie-api:jar:javadoc:7.8.0-SNAPSHOT, org.kie:kie-internal:jar:javadoc:7.8.0-SNAPSHOT: Could not find artifact org.kie:kie-api:jar:javadoc:7.8.0-SNAPSHOT in jboss-public-repository-group (https://repository.jboss.org/nexus/content/groups/public/)
11:45:23     at org.apache.maven.project.DefaultProjectDependenciesResolver.resolve (DefaultProjectDependenciesResolver.java:209)
11:45:23     at info.evanchik.maven.project.SynchronizedProjectDependenciesResolver.resolve (SynchronizedProjectDependenciesResolver.java:42)
11:45:23     at org.apache.maven.lifecycle.internal.LifecycleDependencyResolver.getDependencies (LifecycleDependencyResolver.java:223)
11:45:23     at org.apache.maven.lifecycle.internal.LifecycleDependencyResolver.resolveProjectDependencies (LifecycleDependencyResolver.java:145)
11:45:23     at org.apache.maven.lifecycle.internal.MojoExecutor.ensureDependenciesAreResolved (MojoExecutor.java:246)
11:45:23     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:200)
11:45:23     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
11:45:23     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
11:45:23     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
11:45:23     at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call (MultiThreadedBuilder.java:200)
11:45:23     at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call (MultiThreadedBuilder.java:196)
11:45:23     at java.util.concurrent.FutureTask.run (FutureTask.java:266)
11:45:23     at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:511)
11:45:23     at java.util.concurrent.FutureTask.run (FutureTask.java:266)
11:45:23     at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1149)
11:45:23     at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:624)
11:45:23     at java.lang.Thread.run (Thread.java:748)
11:45:23 Caused by: org.eclipse.aether.resolution.DependencyResolutionException: The following artifacts could not be resolved: org.kie:kie-api:jar:javadoc:7.8.0-SNAPSHOT, org.kie:kie-internal:jar:javadoc:7.8.0-SNAPSHOT: Could not find artifact org.kie:kie-api:jar:javadoc:7.8.0-SNAPSHOT in jboss-public-repository-group (https://repository.jboss.org/nexus/content/groups/public/)
11:45:23     at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies (DefaultRepositorySystem.java:355)
11:45:23     at org.apache.maven.project.DefaultProjectDependenciesResolver.resolve (DefaultProjectDependenciesResolver.java:202)
11:45:23     at info.evanchik.maven.project.SynchronizedProjectDependenciesResolver.resolve (SynchronizedProjectDependenciesResolver.java:42)
11:45:23     at org.apache.maven.lifecycle.internal.LifecycleDependencyResolver.getDependencies (LifecycleDependencyResolver.java:223)
11:45:23     at org.apache.maven.lifecycle.internal.LifecycleDependencyResolver.resolveProjectDependencies (LifecycleDependencyResolver.java:145)
11:45:23     at org.apache.maven.lifecycle.internal.MojoExecutor.ensureDependenciesAreResolved (MojoExecutor.java:246)
11:45:23     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:200)
11:45:23     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
11:45:23     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
11:45:23     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
11:45:23     at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call (MultiThreadedBuilder.java:200)
11:45:23     at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call (MultiThreadedBuilder.java:196)
11:45:23     at java.util.concurrent.FutureTask.run (FutureTask.java:266)
11:45:23     at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:511)
11:45:23     at java.util.concurrent.FutureTask.run (FutureTask.java:266)
11:45:23     at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1149)
11:45:23     at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:624)
11:45:23     at java.lang.Thread.run (Thread.java:748)
11:45:23 Caused by: org.eclipse.aether.resolution.ArtifactResolutionException: The following artifacts could not be resolved: org.kie:kie-api:jar:javadoc:7.8.0-SNAPSHOT, org.kie:kie-internal:jar:javadoc:7.8.0-SNAPSHOT: Could not find artifact org.kie:kie-api:jar:javadoc:7.8.0-SNAPSHOT in jboss-public-repository-group (https://repository.jboss.org/nexus/content/groups/public/)
11:45:23     at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve (DefaultArtifactResolver.java:422)
11:45:23     at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts (DefaultArtifactResolver.java:224)
11:45:23     at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies (DefaultRepositorySystem.java:338)
11:45:23     at org.apache.maven.project.DefaultProjectDependenciesResolver.resolve (DefaultProjectDependenciesResolver.java:202)
11:45:23     at info.evanchik.maven.project.SynchronizedProjectDependenciesResolver.resolve (SynchronizedProjectDependenciesResolver.java:42)
11:45:23     at org.apache.maven.lifecycle.internal.LifecycleDependencyResolver.getDependencies (LifecycleDependencyResolver.java:223)
11:45:23     at org.apache.maven.lifecycle.internal.LifecycleDependencyResolver.resolveProjectDependencies (LifecycleDependencyResolver.java:145)
11:45:23     at org.apache.maven.lifecycle.internal.MojoExecutor.ensureDependenciesAreResolved (MojoExecutor.java:246)
11:45:23     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:200)
11:45:23     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
11:45:23     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
11:45:23     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
11:45:23     at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call (MultiThreadedBuilder.java:200)
11:45:23     at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call (MultiThreadedBuilder.java:196)
11:45:23     at java.util.concurrent.FutureTask.run (FutureTask.java:266)
11:45:23     at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:511)
11:45:23     at java.util.concurrent.FutureTask.run (FutureTask.java:266)
11:45:23     at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1149)
11:45:23     at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:624)
11:45:23     at java.lang.Thread.run (Thread.java:748)
11:45:23 Caused by: org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact org.kie:kie-api:jar:javadoc:7.8.0-SNAPSHOT in jboss-public-repository-group (https://repository.jboss.org/nexus/content/groups/public/)
11:45:23     at org.eclipse.aether.connector.basic.ArtifactTransportListener.transferFailed (ArtifactTransportListener.java:48)
11:45:23     at org.eclipse.aether.connector.basic.BasicRepositoryConnector$TaskRunner.run (BasicRepositoryConnector.java:365)
11:45:23     at org.eclipse.aether.util.concurrency.RunnableErrorForwarder$1.run (RunnableErrorForwarder.java:75)
11:45:23     at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1149)
11:45:23     at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:624)
11:45:23     at java.lang.Thread.run (Thread.java:748)
```

**Comment by jhrcek** [comment]
This seems to be cause by my [disabling javadoc](https://github.com/kiegroup/kie-jenkins-scripts/pull/237) in PR builder jobs. I'm looking into fixing this asap.

File: kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/DMNRuntimeTest.java [hunk 1]
```diff
@@ -2033,5 +2037,26 @@ public void testModelById() {
         assertThat( DMNRuntimeUtil.formatMessages( dmnModel.getMessages() ), dmnModel.hasErrors(), is( false ) );
 
     }
+
+    @Test
+    public void testWeekdayOnDateDMN12() {
```
Comment (baldimir) [inline kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/DMNRuntimeTest.java:2042]: Could you please add some test cases also to this class [1]?

[1] https://github.com/baldimir/drools/blob/611bfacf2c1fc4d5776ca03d7411c992011db2f3/kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/FEELDateTimeDurationTest.java#L141

**Comment by tarilabs** [comment]
(added FEEL-layer test as discussed)

**Review by baldimir — APPROVED** [review]

**Comment by jhrcek** [comment]
Sorry for breaking the build. Here's a proposed fix of PR builder job:
https://github.com/kiegroup/kie-jenkins-scripts/pull/241

**Comment by tarilabs** [comment]
@jhrcek no worries and thanks for the quick fix!

**Comment by tarilabs** [comment]
Jenkins test again

**Pull request merged**